dbms/src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -412,9 +412,19 @@ InterpreterSelectQuery::InterpreterSelectQuery(
             query.setExpression(ASTSelectQuery::Expression::WHERE, std::make_shared<ASTLiteral>(0u));
         need_analyze_again = true;
     }
+    if (query.prewhere() && query.where())
+    {
+        /// Filter block in WHERE instead to get better performance
+        query.setExpression(ASTSelectQuery::Expression::WHERE, makeASTFunction("and", query.prewhere()->clone(), query.where()->clone()));
+        need_analyze_again = true;
```

Comment:
Looks like this optimization removed by #42126 (cc @davenger ) and there is no need in this code anymore, let's try - #54535

Reply:
Hm, actually this is not true, it does not apply PREWHERE always